DOC: update docs for read_csv().na_values and keep_default_na

OXPHOS opened the pull request:
- [x] closes #13967 
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

jreback commented on pandas/io/excel.py:
```diff
@@ -25,6 +25,7 @@
 import pandas.core.common as com
 from warnings import warn
 from distutils.version import LooseVersion
+from parsers import _NA_VALUES
```
add the full import path. Actually we should move these to ``pandas.io.common``.

jreback commented on pandas/io/excel.py:
```diff
@@ -136,16 +137,18 @@ def read_excel(io, sheetname=0, header=0, skiprows=None, skip_footer=0,
+    na_values : str or list-like or dict, default None
+        Additional strings to recognize as NA/NaN. If dict passed, specific
+        per-column NA values. By default the following values are interpreted
+		as NaN: `'""" + "'`, `'".join(sorted(_NA_VALUES)) + """'`.
```
no indent

OXPHOS commented:
Do I need add `what's new` for this? And if so, in which `what's new file`?

Also I am not sure about this error:
```
2016-08-18 18:50:05 (7.84 MB/s) - 'miniconda.sh' saved [21310143/21310143]
ERROR: File or directory already exists: /Users/travis/miniconda
```

jorisvandenbossche commented:
whatsnew is not needed. And don't worry about the travis error, that is something unrelated.

jorisvandenbossche commented on pandas/io/excel.py:
```diff
@@ -138,16 +139,18 @@ def read_excel(io, sheetname=0, header=0, skiprows=None, skip_footer=0,
+    na_values : str or list-like or dict, default None
+        Additional strings to recognize as NA/NaN. If dict passed, specific
+        per-column NA values. By default the following values are interpreted
+        as NaN: `'""" + "'`, `'".join(sorted(_NA_VALUES)) + """'`.
```
I am afraid this does not work with docstrings. You will have to hardcode them like it is done in the `read_csv` docstring

OXPHOS commented:
@jorisvandenbossche Hey I made some changes. Is there anyway I can preview it?

jorisvandenbossche commented:
You just look at the docstring in the console. (with ipython: `pd.read_excel?`)

jreback commented on pandas/io/excel.py:
```diff
@@ -74,100 +168,13 @@ def get_writer(engine_name):
                date_parser=None, na_values=None, thousands=None,
                convert_float=True, has_index_names=None, converters=None,
                engine=None, squeeze=False, **kwds):
-    """
```
I don't think its necessary to make this an Appender doc-string. We generally only do that if you need it more than once. @jorisvandenbossche ?

jorisvandenbossche replied:
Well, it's to be able to automatically inject the NA_VALUES instead of manually writing it in the docstring. 

It's a bit a trade-off here: is the added complexity of moving the docstring away from the actual function worth it for the 1 line of na values that is written manually (and has to be updated if the default recognized NaN values are changed).

jreback replied:
oh I c. Then this is ok. We want to have the common NA values in a single place

jorisvandenbossche commented on pandas/io/excel.py:
```diff
@@ -27,12 +28,105 @@
+na_values : str or list-like or dict, default None
+    Additional strings to recognize as NA/NaN. If dict passed, specific
+    per-column NA values. By default the following values are interpreted
+    as NaN: `'""" + "'`, `'".join(sorted(_NA_VALUES)) + """'`.
```
I think the backticks are not needed around the strings. For the rest, the docstring looks fine in the console!

OXPHOS committed:
GH13967: move around _NA_VALUES and add doc for read_csv().na_values

OXPHOS commented:
I squashed and removed the backticks.

I did test:
```py
In [12]: _NA_VALUES = set([
   ....:     '-1.#IND', '1.#QNAN', '1.#IND', '-1.#QNAN', '#N/A N/A', '#N/A',
   ....:     'N/A', 'NA', '#NA', 'NULL', 'NaN', '-NaN', 'nan', '-nan', ''
   ....: ])

In [13]: s = """as NaN: '""" + "', '".join(sorted(_NA_VALUES)) + """'."""

In [14]: print s
as NaN: '', '#N/A', '#N/A N/A', '#NA', '-1.#IND', '-1.#QNAN', '-NaN', '-nan', '1.#IND', '1.#QNAN', 'N/A', 'NA', 'NULL', 'NaN', 'nan'.
```

I didn't check the doc with installation and `pd.read_excel?` because I got a [weird error](http://pastebin.com/kBHCAHbc) out of nowhere. I am trying to fix it but I have no clue how this is happening yet. So should I post the problem in anywhere if I have no luck hacking it? Like google group or irc channel? Thanks!

OXPHOS commented:
Okay I fixed it. Something's wrong with Cython. 
Ready to merge!

jreback commented:
thanks!

Can you do a followup where we make this wrap on the NaNs (they generate an extra long line)
```
na_values : str or list-like or dict, default None
    Additional strings to recognize as NA/NaN. If dict passed, specific
    per-column NA values. By default the following values are interpreted
    as NaN: '', '#N/A', '#N/A N/A', '#NA', '-1.#IND', '-1.#QNAN', '-NaN', '-nan', '1.#IND', '1.#QNAN', 'N/A', 'NA', 'NULL', 'NaN', 'nan'.
thousands : str, default None
    Thousands separator for parsing string columns to numeric.  Note that
    this parameter is only necessary for columns stored as TEXT in Excel,
    any numeric columns will automatically be parsed, regardless of display
    format.
```

OXPHOS commented:
Sure I'll try!
